File: frontend/src/types/match.ts
import { CheckinItem } from "./checkin";

export interface MatchResponse {
  team_a: CheckinItem[]; // Lista de CheckinItems
  team_b: CheckinItem[]; // Lista de CheckinItems
  waiting_team_1: CheckinItem[];
  waiting_team_2: CheckinItem[];
  following_list: CheckinItem[]; // Lista de CheckinItems
  can_randomize: boolean | null;
  can_choose_draw: boolean | null;
  match_time_rule: string;
  next_up: string | null; // Pode ser null
}

File: backend/app/schemas/match.py
from pydantic import BaseModel
from typing import Optional, List
#from datetime import datetime
from enum import Enum
from app.schemas.checkin import CheckinResponse

class MatchResultEnum(str, Enum):
    losing_team_a = "team_a"
    losing_team_b = "team_b"
    draw = "draw"

class MatchEndRequest(BaseModel):
    result: MatchResultEnum

class MatchStateResponse(BaseModel):
    team_a: List[CheckinResponse]
    team_b: List[CheckinResponse]
    waiting_team_1: List[CheckinResponse]
    waiting_team_2: List[CheckinResponse]
    following_list: List[CheckinResponse]
    can_randomize: Optional[bool] = False
    can_choose_draw: Optional[bool] = False

    match_time_rule: Optional[str] = None

    next_up: Optional[str] = None